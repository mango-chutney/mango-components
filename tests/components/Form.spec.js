// @flow

import 'raf/polyfill';
import 'jest-styled-components';
import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Field, reduxForm, reducer as formReducer } from 'redux-form';
import renderer from 'react-test-renderer';
import MC from '../../src';

const Input = MC.Input.createComponent();
const TextArea = MC.TextArea.createComponent();
const Checkbox = MC.Checkbox.createComponent();
const PaddedCheckbox = MC.PaddedCheckbox.createComponent();

const store = createStore(combineReducers({ form: formReducer }));

function FormComponents() {
  return (
    <form>
      <Field
        type="text"
        name="input"
        placeholder="Input"
        label="Input"
        component={Input}
      />
      <Field
        name="textarea"
        placeholder="Text Area"
        label="Text Area"
        component={TextArea}
      />
      <Field
        name="checkbox"
        label="Checkbox"
        placeholder="Checkbox"
        component={Checkbox}
      />
      <Field
        name="padded-checkbox"
        label="PaddedCheckbox"
        placeholder="PaddedCheckbox"
        component={PaddedCheckbox}
      />
    </form>
  );
}

const Form = reduxForm({
  form: 'FormComponents',
})(FormComponents);

it('Form components (Input, TextArea, Checkbox, PaddedCheckbox) renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Form />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
