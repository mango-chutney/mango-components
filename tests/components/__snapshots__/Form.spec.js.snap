// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form components (Input, TextArea, Checkbox, PaddedCheckbox) renders correctly 1`] = `
.c6 {
  position: relative;
  margin: 0;
  height: 1.25rem;
  width: 1.25rem;
  display: inline-block;
}

.c7 {
  width: 100%;
  height: 100%;
  position: absolute;
  padding: 0;
  margin: 0;
  top: 0;
  left: 0;
  z-index: 9001;
  opacity: 0;
}

.c7[disabled] {
  cursor: not-allowed;
}

.c7:checked + .c8::before,
.c7:checked + .c8::after {
  top: 0;
  left: 0;
}

.c7:checked + .c8::before {
  background: #2ea1f8;
  border-color: #1596f7;
  -webkit-transition: background-color 0.3s;
  transition: background-color 0.3s;
}

.c7:checked + .c8::after {
  content: '';
  background-image: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0naHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmcnIHhtbG5zOnhsaW5rPSdodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rJyBwcmVzZXJ2ZUFzcGVjdFJhdGlvPSd4TWlkWU1pZCcgd2lkdGg9JzE2LjUnIGhlaWdodD0nMTUnIHZpZXdCb3g9JzAgMCAxNi41IDE1Jz48cGF0aCBmaWxsPScjZmZmJyBmaWxsLXJ1bGU9J2V2ZW5vZGQnIGQ9J00xNi41MTUsMS4wNzEgQzE2LjM5OSwwLjgyNyAxNS45MTQsLTAuMDMwIDE1LjUxNSwwLjAwMSBDMTQuMjg5LDAuNzY1IDEyLjY5MywyLjM1NCAxMS42MzgsMy4zNjIgQzkuNTI4LDUuMzc4IDcuNTAzLDcuNDg2IDUuNTA4LDkuNjI0IEwxLjUxNiw1Ljg5NyBMMC4wMDUsNy40ODYgQzIuMjAwLDkuODM3IDQuNTY3LDEyLjM3MyA2LjUwNSwxNS4wMDAgQzguOTg2LDEwLjYzMiAxMy4wMzUsNC4zNzAgMTYuNTE1LDEuMDcxIEwxNi41MTUsMS4wNzEgWicvPjwvc3ZnPg==');
  background-position: 50% 40%;
  background-repeat: no-repeat;
  background-size: 70%;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c9 {
  cursor: pointer;
  display: inline-block;
  min-height: 1.25rem;
  line-height: 1.25rem;
  position: relative;
  border: 1px solid transparent;
  display: block;
}

.c9::before,
.c9::after {
  width: 1.25rem;
  height: 1.25rem;
  cursor: pointer;
  content: '';
  display: inline-block;
  border-radius: 4px;
  vertical-align: middle;
}

.c9::before {
  border: 1px solid #e2e7ee;
  background: #fafbfc;
  margin-right: 1rem;
}

.c9::after {
  position: absolute;
  left: 0;
  border: 1px solid transparent;
  line-height: 1.25rem;
  margin-top: 0.125rem;
  text-align: center;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #fafbfc;
  border-color: #e2e7ee;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  color: #354052;
  display: block;
  font-family: proxima-nova,Helvetica Neue Light,Helvetica Neue,Helvetica,Arial,sans-serif;
  height: 2.6rem;
  margin-bottom: 1rem;
  outline: 0;
  padding: 0.5rem 1rem;
  -webkit-transition: border-color 300ms ease;
  transition: border-color 300ms ease;
  width: 100%;
}

.c2::-webkit-input-placeholder {
  color: rgba(127,143,164,0.8);
}

.c2::-moz-placeholder {
  color: rgba(127,143,164,0.8);
}

.c2:-ms-input-placeholder {
  color: rgba(127,143,164,0.8);
}

.c2::placeholder {
  color: rgba(127,143,164,0.8);
}

.c2:active,
.c2:focus {
  border-color: #354052;
}

.c1 {
  display: block;
  position: relative;
}

.c1::after {
  content: '';
  color: rgba(127,143,164,0.8);
  font: normal normal normal 0.875rem tristicons;
  line-height: 1rem;
  position: absolute;
  right: 1rem;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c0 {
  font-size: 0.875rem;
  font-weight: 600;
  display: block;
}

.c10 {
  font-size: 0.875rem;
  font-weight: 600;
  display: block;
  display: inline-block;
  margin-left: 1rem;
  font-weight: normal;
}

.c5 {
  margin: 1rem 0;
}

.c11 {
  display: block;
  position: relative;
}

.c11::after {
  content: '';
  color: rgba(127,143,164,0.8);
  font: normal normal normal 0.875rem tristicons;
  line-height: 1rem;
  position: absolute;
  right: 1rem;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c11::after {
  content: "\\f10a";
}

.c12 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #fafbfc;
  border-color: #e2e7ee;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  color: #354052;
  display: block;
  font-family: proxima-nova,Helvetica Neue Light,Helvetica Neue,Helvetica,Arial,sans-serif;
  height: 2.6rem;
  margin-bottom: 1rem;
  outline: 0;
  padding: 0.5rem 1rem;
  -webkit-transition: border-color 300ms ease;
  transition: border-color 300ms ease;
  width: 100%;
}

.c12::-webkit-input-placeholder {
  color: rgba(127,143,164,0.8);
}

.c12::-moz-placeholder {
  color: rgba(127,143,164,0.8);
}

.c12:-ms-input-placeholder {
  color: rgba(127,143,164,0.8);
}

.c12::placeholder {
  color: rgba(127,143,164,0.8);
}

.c12:active,
.c12:focus {
  border-color: #354052;
}

.c12::-ms-expand {
  display: none;
}

.c12 option {
  width: 100%;
}

.c3 {
  display: block;
  position: relative;
}

.c3::after {
  content: '';
  color: rgba(127,143,164,0.8);
  font: normal normal normal 0.875rem tristicons;
  line-height: 1rem;
  position: absolute;
  right: 1rem;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c3::after {
  display: none;
}

.c4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #fafbfc;
  border-color: #e2e7ee;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  color: #354052;
  display: block;
  font-family: proxima-nova,Helvetica Neue Light,Helvetica Neue,Helvetica,Arial,sans-serif;
  height: 2.6rem;
  margin-bottom: 1rem;
  outline: 0;
  padding: 0.5rem 1rem;
  -webkit-transition: border-color 300ms ease;
  transition: border-color 300ms ease;
  width: 100%;
  height: auto;
  padding: 1.25rem 1rem;
  resize: vertical;
}

.c4::-webkit-input-placeholder {
  color: rgba(127,143,164,0.8);
}

.c4::-moz-placeholder {
  color: rgba(127,143,164,0.8);
}

.c4:-ms-input-placeholder {
  color: rgba(127,143,164,0.8);
}

.c4::placeholder {
  color: rgba(127,143,164,0.8);
}

.c4:active,
.c4:focus {
  border-color: #354052;
}

<form>
  <label
    className="c0"
    htmlFor="input"
    placeholder="Input"
    type="text"
  >
    Input
    <span
      className="c1"
    >
      <input
        className="c2"
        id="input"
        name="input"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Input"
        type="text"
        value=""
      />
    </span>
  </label>
  <label
    className="c0"
    htmlFor="textarea"
    placeholder="Text Area"
  >
    Text Area
    <span
      className="-sc-bxivhb c3"
    >
      <textarea
        className="-textarea-sc-ifAKCX c4"
        id="textarea"
        name="textarea"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Text Area"
        value=""
      />
    </span>
  </label>
  <div
    className="c5"
  >
    <div
      className="c6"
    >
      <input
        className="c7"
        id="checkbox"
        name="checkbox"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Checkbox"
        type="checkbox"
        value=""
      />
      <div
        className="c8 input-decorator c9"
      />
    </div>
    <label
      className="-sc-bdVaJa c10"
      htmlFor="checkbox"
      placeholder="Checkbox"
    >
      Checkbox
    </label>
  </div>
  <label
    className="c0"
    htmlFor="select"
    placeholder="select"
  >
    select
    <span
      className="-sc-bwzfXH c11"
    >
      <select
        className="-select-sc-htpNat c12"
        id="select"
        name="select"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="select"
        value=""
      >
        <option
          value="option-1"
        >
          Option 1
        </option>
        <option
          value="option-2"
        >
          Option 2
        </option>
      </select>
    </span>
  </label>
</form>
`;
